### ISSUE TITLE
Add a possibility to automatically hide shares from the stream, which already have been presented in the stream by somebody else

### ISSUE THREAD

**Issue by cmrd-senya** [diaspora/diaspora#6273]
In some other social networks they show shares in your stream only once, and if somebody of your friends shared a post of your other friend, you won't see it. So it prevents your stream from flooding, where your friends circle is close and they share each others posts often. 

Lets implement it at least as an option.

**Comment by Flaburgan** [comment]
Really close to https://github.com/diaspora/diaspora/issues/4219, I think I would prefer to see this one implemented.

**Comment by jhass** [comment]
I think this is a duplicate of #4219. If you disagree please elaborate (on the differences) and we can reopen.
